### PR TITLE
#170180280 Manager get all and specific user booking  Requests

### PR DESCRIPTION
**Pull request by william000000**
#### What does this PR do?
 Manager get specific or all booking accommodation requests
#### Description of Task to be completed?
- Get specific booking accommodation requests 
- Get all users' booking requests
- Fix the issue about the user who made the trip to be the one to book an accommodation
- Add tests
- Add new fields in booking tables `userId`
- Swagger documentation 
#### How should this be manually tested?
- You have to clone the repo `git clone https://github.com/andela/team-odd-bn-backend.git`
- Then After the repo cloned `cd project-folder` and run `npm i` to install all dependencies
- Go to my branch `git checkout ft-get-all-and-specific-booking-request-170180280`
- Insert the data by `npx sequelize-cli db:seed:all`
- Run app by `npm run dev-start`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170180280](https://www.pivotaltracker.com/story/show/170180280)
##### Get all users' booking requests
image
##### Get specific user booking requests
image

#### Questions:
N/A

### CODE REVIEW

File: src/services/BookingService.js [hunk 1]
```diff
@@ -60,20 +60,84 @@ class BookingService {
...
+          model: tripRequests,
+          where: { userId: id }
+        }],
+        raw: true
+      };
+      const result = await CommonQueries.findAll(trips, findOneBookingObj);
+
+      return result;
+    } catch (error) {
+      console.log(error);
```
Comment (NiyoEric) [inline src/services/BookingService.js]: Remove the console.

File: src/tests/100-bookingTest.js [hunk 2]
```diff
@@ -119,3 +128,42 @@ describe('Book an accomadition facility', () => {
...
+        res.body.should.be.an('object');
+        done(err);
+      });
+  });
+  it('should not be able to get a specific user booking request when the request does not found', (done) => {
+    chai.request(app).get('/api/v1/trips/booking/8000')
+      .set('token', managerSignin)
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.an('object');
```
Comment (NiyoEric) [inline src/tests/100-bookingTest.js:166]: Add more test.

File: src/tests/100-bookingTest.js [hunk 3]
```diff
@@ -119,3 +128,42 @@ describe('Book an accomadition facility', () => {
       });
   });
 });
+describe('Get User Booking accomadition request', () => {
+  it('should be able to get user booking request', (done) => {
+    chai.request(app).get('/api/v1/trips/booking')
+      .set('token', managerSignin)
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.an('object');
```
Comment (NiyoEric) [inline src/tests/100-bookingTest.js:134]: Please add more test.

**Review by NiyoEric — CHANGES_REQUESTED** [review]
Well done @william000000. I liked how you made it so simple to display all bookings. I request you add more test.

File: src/tests/100-bookingTest.js (hunk 3 above)
Comment (william000000) [reply src/tests/100-bookingTest.js:134]: Thank you @NiyoEric, I have already implemented

**Review by NiyoEric — APPROVED** [review]

File: src/controllers/BookingController.js [hunk 4]
```diff
@@ -23,5 +23,24 @@ class BookingController {
       return Response.errorMessage(req, res, 'Server error', 500);
     }
   }
+
+  /**
+ * users can book an accommodation facility
+ * @static
```
Comment (victkarangwa) [inline src/controllers/BookingController.js]: I think it should be like this:
```js
  /**
 * Manager can get all booking requests
```

File: src/routes/api/bookingRoute.js [hunk 5]
```diff
@@ -70,5 +71,51 @@ const { isTripFound, isRoomBooked } = Conflict;
 
 bookingRouter.post('/:tripId/booking', verifyToken, isUserVerified, isTripFound, Validate.bookAccommodationRules(), checkInputDataError, checkIfCheckInDateIsAsCheckOutDate, isRoomBooked, AccommodationMiddleware.isAccommodationAvail, BookingController.bookAccommodation);
 
+/**
+ * @swagger
+ *
+ * /trips/booking:
+ *    get:
+ *      summary: User can book an accommodation
+ *      tags: [Booking]
```
Comment (victkarangwa) [inline src/routes/api/bookingRoute.js]: Let's just modify the description to their corresponding feature functionality
```js
 *    get:
 *      summary: Manager can get all bookings
 *      tags: [Booking]
```

File: src/routes/api/bookingRoute.js [hunk 6]
```diff
@@ -70,5 +71,51 @@ const { isTripFound, isRoomBooked } = Conflict;
...
+ */
+
+bookingRouter.get('/booking', verifyToken, isUserVerified, isManagerHasAccess, BookingController.getUserBookingRequest);
+/**
+ * @swagger
+ *
+ * /trips/booking/{userId}:
+ *    get:
+ *      summary: User can book an accommodation
+ *      tags: [Booking]
```
Comment (victkarangwa) [inline src/routes/api/bookingRoute.js]:
```js
 *    get:
 *      summary: Manager can get a specific booking
 *      tags: [Booking]
```

**Review by victkarangwa — COMMENTED** [review]
Nice job @william000000 
Just have a look at some feedback I left

**Review by hezronkimutai — COMMENTED** [review]
Nice work @william000000 .

**Review by hezronkimutai — APPROVED** [review]

File: src/middlewares/Validate.js [hunk 7]
```diff
@@ -258,6 +258,7 @@ class Validate {
     const day = d.getDate();
     const correctDate = new Date(year, month, day).toDateString();
     return [
+      check('tripId', 'tripId should be an integer').isNumeric(),
```
Comment (hezronkimutai) [inline src/middlewares/Validate.js]: isNumeric() method accepts integers, decimals and negative numbers. I suggest using isInt() method for validating only integers.

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
Hi @william000000 , nice job.

**Review by victkarangwa — APPROVED** [review]

**Review by nziokaivy — APPROVED** [review]
Great job @william000000 Looks good to me!

**Review by hezronkimutai — APPROVED** [review]

**Review by julietezekwe — APPROVED** [review]

**Pull request merged**